### PR TITLE
HOTFIX: Check for "underpriced transaction" indicating that nonce is too low

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

**Comment by j16r** [comment]
Do a merge of master into this branch to fix the conflict 👍

**Comment by samsondav** [comment]
@j16r do you mean develop? master is already merged

File: core/store/models/log_events.go [hunk 1]
```diff
@@ -482,8 +489,16 @@ func (parseRunLog20190207withoutIndexes) parseJSON(log eth.Log) (JSON, error) {
 	dataLengthStart := expirationEnd + versionSize + dataLocationSize
 	cborStart := dataLengthStart + dataLengthSize
 
+	if len(log.Data) < dataLengthStart+32 {
+		return JSON{}, errors.New("malformed data")
+	}
```
Comment (felder-cl) [inline core/store/models/log_events.go]: not-blocker: where does the 32 come from?

File: core/store/models/log_events.go [hunk 2]
```diff
@@ -482,8 +489,16 @@ func (parseRunLog20190207withoutIndexes) parseJSON(log eth.Log) (JSON, error) {
...
+
 	dataLength := whisperv6.BytesToUintBigEndian(data[dataLengthStart : dataLengthStart+32])
 
+	if len(log.Data) < cborStart+int(dataLength) {
+		return JSON{}, errors.New("cbor too short")
+	}
```
Comment (felder-cl) [inline core/store/models/log_events.go]: integer overflow on 32 bit systems: positive -> negative.  Is this checked somewhere?

File: core/store/tx_manager.go [hunk 3]
```diff
@@ -230,7 +236,7 @@ func (txm *EthTxManager) createTx(
 	gasLimit uint64,
 	value *assets.Eth) (*models.Tx, error) {
 
-	for nrc := 0; nrc <= nonceReloadLimit; nrc++ {
+	for nrc := 0; nrc < nonceReloadLimit+1; nrc++ {
```
Comment (felder-cl) [inline core/store/tx_manager.go]: not-a-blocker: where does the +1 come from?

**Pull request closed without merging**